### PR TITLE
Use SIGTERM to end test process for yaml tests

### LINKED ISSUE #31574: [Tooling] Adjust run_test_suite.py to no t use SIGKILL on the test device
### Reproduction steps / Feature

I'm currently changing the matter.js testing to use the new tooling from here and I run into the issue that it seems that the started Device executable is ended by SIGKILL or such "hard way" when it gets restarted. 
I use a nodejs script and use this e.g. as --all-clusters-app parameter, but SIGKILL do not give the process time to really tear down anything. e.g. sending MDNS expires and such and this leads to issues with the tests.

It would be very helpful of the kill signal could be sonfigurable - or just SIGTERM or such is used.

Thank you

### Platform

core (please add to version below)

### Platform Version(s)

NA

### Type

YAML tested

### (Optional) If manually tested please explain why this is only manually tested

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by Apollon77**
fixes #31574

I tested it with chip all clusters-app and also matter.js and it works, also the case where it falls back to killing if process did not terminates after 10s

I also verified that this solves the testing error I have with the TestDiscovery test which was caused by missing MDNS expires

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

File: scripts/tests/chiptest/test_definition.py [hunk 1]
```diff
@@ -158,6 +155,18 @@ def __updateSetUpCode(self):
...
 
+    def __terminateProcess(self):
+        if self.process:
+            self.process.terminate()  # sends SIGTERM
+            try:
+                self.process.wait(10)
+            except subprocess.TimeoutExpired:
+                logging.debug('Subprocess did not terminated on SIGTERM, killing it now')
+                self.process.kill()
+                self.process.wait(10)
```
Comment (tehampson) [inline scripts/tests/chiptest/test_definition.py:166]: nit, Why are we waiting for 10 here? Can we have comment on why magic value was selected for after killing?

**Review by tehampson — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

File: scripts/tests/chiptest/test_definition.py (hunk 1 above)
Comment (Apollon77) [reply scripts/tests/chiptest/test_definition.py:166]: Please solve the conversion if ok or tell me what I should change. Thank you

**Comment by andy31415** [comment]
@Apollon77 you do not need to merge with master that frequently - that resets CI and it is very expensive (2h CI).

**Comment by Apollon77** [comment]
Yea I already saw that and decided. The wait till tomorrow ;-) there is an answer open anyway.

File: scripts/tests/chiptest/test_definition.py [hunk 2]
```diff
@@ -158,6 +155,18 @@ def __updateSetUpCode(self):
             raise Exception("Unable to find QR code")
         self.setupCode = qrLine.group(1)
 
+    def __terminateProcess(self):
+        if self.process:
+            self.process.terminate()  # sends SIGTERM
+            try:
+                self.process.wait(10)
+            except subprocess.TimeoutExpired:
+                logging.debug('Subprocess did not terminated on SIGTERM, killing it now')
```
Comment (bzbarsky-apple) [inline scripts/tests/chiptest/test_definition.py:164]: "did not terminate", not "did not terminated", @Apollon77

Comment (Apollon77) [reply scripts/tests/chiptest/test_definition.py:164]: Uups. Right typo. Will provide a fix PR later.

Comment (Apollon77) [reply scripts/tests/chiptest/test_definition.py:164]: @bzbarsky-apple see https://github.com/project-chip/connectedhomeip/pull/31702

Comment (bzbarsky-apple) [reply scripts/tests/chiptest/test_definition.py:164]: Thank you!

**Pull request merged**